common/src/main/java/com/datastax/oss/common/sink/config/CassandraSinkConfig.java

```diff
@@ -347,6 +352,32 @@ private void deprecatedSecureBundle(Map<String, String> connectorSettings) {
         Function.identity());
   }
 
+  static void decodeBase64EncodedSecureBundle(Map<String, String> javaDriverSettings) {
+    // it is very awkward on k8s to pass ZIP files
+    // if the path is base64:xxxx we decode the payload and create
+    // a temporary file
+    // we are setting permissions such that only current user can access the file
+    // the file will be deleted at JVM exit
+    String encoded = javaDriverSettings.get(SECURE_CONNECT_BUNDLE_DRIVER_SETTING);
```

Comment:
this change is non-backward compatible. What about the clients that are using this parameter in the old way?
We should make it backward compatible. One way would be to fall back to the old way if the `! encoded.startsWith("base64:")`. The other is to create a new parameter.

Reply:
@tomekl007 
I am not sure I follow, at line 362 we are checking that the string starts with "base64:"

in fact all of the integration tests are passing, both the ones using a file path and an HTTP URL

Reply:
ah ok, this is executed after the initial settings are propagated to the java driver settings via `SECURE_CONNECT_BUNDLE_DRIVER_SETTING`. If encoding is present then it overrides that, right? That was the main intention?

Reply:
yes it is